src/pico/i_video.c

```diff
@@ -949,108 +797,102 @@ void __noinline new_frame_init_overlays_palette_and_wipe() {
+        new_frame_init_overlays_palette_and_wipe();
+    }
+
+
 #if USE_INTERP
     need_save = interp_in_use;
     interp_updated = 0;
 #endif
+    uint16_t buffer[SCREENWIDTH];
+    I_handleFrameStart(frame);
 
-    while (buffer) {
-        static int8_t last_frame_number = -1;
-        int frame = scanvideo_frame_number(buffer->scanline_id);
-        int scanline = scanvideo_scanline_number(buffer->scanline_id);
-        if ((int8_t) frame != last_frame_number) {
-            last_frame_number = frame;
-            new_frame_stuff();
-        }
+    const uint8_t end_scanline = SUPPORT_OVERLAYS ? SCREENHEIGHT : MAIN_VIEWHEIGHT;
 
+    for (int scanline = 0; scanline < end_scanline; scanline++){
         DEBUG_PINS_SET(scanline_copy, 1);
-        if (display_video_type != VIDEO_TYPE_TEXT) {
-            // we don't have text mode -> normal transition yet, but we may for network game, so leaving this here - we would need to put the buffer pointers back
-            assert (buffer->data < text_scanline_buffer_start || buffer->data >= text_scanline_buffer_start + TEXT_SCANLINE_BUFFER_TOTAL_WORDS);
-            scanline_funcs[display_video_type](buffer->data+1, scanline);
-            if (display_video_type >= FIRST_VIDEO_TYPE_WITH_OVERLAYS) {
-                assert(scanline < count_of(vpatchlists->vpatch_starters));
-                int prev = 0;
-                for (int vp = vpatchlists->vpatch_starters[scanline]; vp;) {
-                    int next = vpatchlists->vpatch_next[vp];
-                    while (vpatchlists->vpatch_next[prev] && vpatchlists->vpatch_next[prev] < vp) {
-                        prev = vpatchlists->vpatch_next[prev];
-                    }
-                    assert(prev != vp);
-                    assert(vpatchlists->vpatch_next[prev] != vp);
-                    vpatchlists->vpatch_next[vp] = vpatchlists->vpatch_next[prev];
-                    vpatchlists->vpatch_next[prev] = vp;
-                    prev = vp;
-                    vp = next;
-                }
-                vpatchlist_t *overlays = vpatchlists->overlays[display_overlay_index];
-                prev = 0;
-                for (int vp = vpatchlists->vpatch_next[prev]; vp; vp = vpatchlists->vpatch_next[prev]) {
-                    patch_t *patch = resolve_vpatch_handle(overlays[vp].entry.patch_handle);
-                    int yoff = scanline - overlays[vp].entry.y;
-                    if (yoff < vpatch_height(patch)) {
-                        vpatchlists->vpatch_doff[vp] = draw_vpatch((uint16_t*)(buffer->data + 1), patch, &overlays[vp],
-                                                                   vpatchlists->vpatch_doff[vp]);
-                        prev = vp;
-                    } else {
-                        vpatchlists->vpatch_next[prev] = vpatchlists->vpatch_next[vp];
-                    }
-                }
-            }
-            uint16_t *p = (uint16_t *) buffer->data;
-            p[0] = video_doom_offset_raw_run;
```

Comment:
This stuff is VGA specific and crops up in a few places, so we just get rid of it